/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package tflite.task.audio;

import "tensorflow_lite_support/cc/task/core/proto/base_options.proto";

// Options for setting up an AudioClassifier.
// Next Id: 7
message AudioClassifierOptions {
  // Base options for configuring the external model file.
  optional tflite.task.core.BaseOptions base_options = 1;

  // The locale to use for display names specified through the TFLite Model
  // Metadata, if any. Defaults to English.
  optional string display_names_locale = 2 [default = "en"];

  // The maximum number of top-scored classification results to return. If < 0,
  // all available results will be returned. If 0, an invalid argument error is
  // returned.
  optional int32 max_results = 3 [default = -1];

  // Score threshold, overrides the ones provided in the model metadata
  // (if any). Results below this value are rejected.
  optional float score_threshold = 4;

  // Optional allowlist of class names. If non-empty, classifications whose
  // class name is not in this set will be filtered out. Duplicate or unknown
  // class names are ignored. Mutually exclusive with class_name_denylist.
  repeated string class_name_allowlist = 5;

  // Optional denylist of class names. If non-empty, classifications whose
  // class name is in this set will be filtered out. Duplicate or unknown
  // class names are ignored. Mutually exclusive with class_name_allowlist.
  repeated string class_name_denylist = 6;
}
